src/algorithm/hnswlib/visited_list_pool.h

```diff
@@ -102,7 +102,7 @@ class VisitedListPool {
     void
     releaseVisitedList(VisitedListPtr vl) {
         std::unique_lock<std::mutex> lock(poolguard_);
-        pool_.push_back(vl);
+        pool_.emplace_back(vl);
     }
```

Comment:
push back and  pop back means the vl at front may not be used

Reply:
All we need to do is ensure that after we retrieve a visit list, we remove it from the queue. We don't need to worry about the position to put it back in later.